src/deadline/client/cli/_groups/bundle_group.py

```diff
@@ -81,6 +83,85 @@ def validate_parameters(ctx, param, value):
     return parameters_split
 
 
+def _validate_submitter_info(ctx, param, value):
+    """
+    Validate provided --submitter-info value and convert to SubmitterInfo object.
+
+    Supports two input formats (mutually exclusive):
+    - Multiple key=value pairs: --submitter-info submitter_name=MyApp --submitter-info host_application_name=Maya
+    - Single inline JSON blurb: --submitter-info '{"submitter_name": "MyApp", "additional_info": {"custom": "data"}}'
+
+    All keys must be valid SubmitterInfo fields. Unknown keys will raise an error.
+    """
+    if not value:
+        return None
+
+    # Get valid field names from SubmitterInfo dataclass
+    valid_fields = {field.name for field in fields(SubmitterInfo)}
+
+    # Check if using JSON format (only one value allowed and it must start with {)
+    if len(value) == 1 and value[0].startswith("{"):
+        # Inline JSON format
+        try:
+            data = json.loads(value[0])
+            if not isinstance(data, dict):
+                raise click.BadParameter("Inline JSON submitter info must be an object")
+        except json.JSONDecodeError as e:
+            raise click.BadParameter(f"Invalid inline JSON for submitter info: {e}") from e
+
+        # Validate all keys
+        for key in data.keys():
+            if key not in valid_fields:
+                raise click.BadParameter(
+                    f"Unknown field '{key}' in inline JSON. "
+                    f"Valid fields are: {', '.join(sorted(valid_fields))}"
+                )
+
+        # Validate additional_info if present
+        if "additional_info" in data and not isinstance(data["additional_info"], dict):
+            raise click.BadParameter("additional_info must be an object")
+
+        info_dict = data
+
+    else:
+        # key=value format - check that none start with {
+        for item in value:
+            if item.startswith("{"):
+                raise click.BadParameter(
+                    "Cannot specify multiple inline JSON values or both inliune JSON and key=value pairs. "
+                    "Use either a single JSON object OR multiple key=value pairs."
+                )
+
+        # Parse key=value pairs
+        info_dict = {}
+        for item in value:
+            match = re.match(r"([^=]+)=(.*)", item)
+            if not match:
+                raise click.BadParameter(
+                    f'Submitter info must be in format "key=value" Invalid: {item}'
+                )
+
+            key, val = match.groups()
+            if key not in valid_fields:
+                raise click.BadParameter(
+                    f"Unknown field '{key}'. Valid fields are: {', '.join(sorted(valid_fields))}"
+                )
+
+            info_dict[key] = val
```

Comment:
nit: we can simplify this a little, but totally optional:

```suggestion
            if '=' not in item:
                raise click.BadParameter(
                    f'Submitter info must be in format "key=value" Invalid: {item}'
                )

            key, val = item.split('=', maxsplit=1)
            if key not in valid_fields:
                raise click.BadParameter(
                    f"Unknown field '{key}'. Valid fields are: {', '.join(sorted(valid_fields))}"
                )

            info_dict[key] = val
```

Otherwise we will want to anchor the match:

```suggestion
            match = re.match(r"^([^=]+)=(.*)$", item)
            if not match:
                raise click.BadParameter(
                    f'Submitter info must be in format "key=value" Invalid: {item}'
                )

            key, val = match.groups()
            if key not in valid_fields:
                raise click.BadParameter(
                    f"Unknown field '{key}'. Valid fields are: {', '.join(sorted(valid_fields))}"
                )

            info_dict[key] = val
```

Reply:
Ended up taking a suggestion from a comment Mia had https://github.com/aws-deadline/deadline-cloud/pull/940#discussion_r2603869511. Copied the logic we use in the openjd CLI to clean things up a bit and now leaves the option in the future to add the same for parsing the job parameters.